clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -225,22 +226,27 @@ public static Properties consumerConfig(final String bootstrapServers, Class key
     }
 
     /**
-     *  uses default value of 30 seconds for timeout
+     *  uses default value of 15 seconds for timeout, and default message for failure
      */
     public static void waitForCondition(TestCondition testCondition) throws InterruptedException {
-        waitForCondition(testCondition, 30000);
+        waitForCondition(testCondition, 15000, "Expected condition not met by timeout");
```

Comment:
Wouldn't it be better to also take a message in this method? I don't think we ever want a generic timeout error. One way to make it more concise here and in the other method too is to change the semantics of the string passed as a parameter. Instead of it being the full error message, it could be the details only. So, when constructing the assertion failure, we would say something like:

```java
"Condition not met within timeout " + maxTimeMillis + ". " + conditionDetails
```

Thoughts?